docs/dev-guide/guide.md

```diff
@@ -471,43 +471,61 @@ FOR NO KEY UPDATE;
 
 ## Safely interacting with the database
 
-??? note
+!!! note
 
     This pattern is currently being rolled out as a gradual refactor of existing code on a model-by-model basis.
 
-Almost every page is dealing with database data, so it is important to understand how to interact with the database securely. Data in `res.locals` has already been validated and is safe to use. However, it is recommended to type this data using the `page-context` helpers for extra type safety. For example,
+Almost every page is dealing with database data, so it is important to understand how to interact with the database securely.
 
-```typescript
-const { authz_data } = getPageContext(res.locals);
-const { course_instance: courseInstance } = getCourseInstance(res.locals, 'student');
-```
-
-For most API/POST handlers, we want to lookup or modify data based on unvalidated query parameters or request body fields. It is easy to forget to validate these fields with the correct authorization levels. To help with this, we are moving to a pattern where model functions should accept full, typed row objects as parameters. For example,
+For most API/POST handlers, we want to lookup or modify data based on unvalidated query parameters or request body fields. It is easy to forget to validate these fields with the correct authorization levels. To help with this, we are moving to a pattern where model functions should accept full, typed row objects as parameters. As an example, we want to make it hard to update an enrollment status using just an enrollment ID.
 
 ```typescript
-await updateEnrollmentStatus({
-  enrollment: my_enrollment,
-  status: 'joined',
+// We want to prove that we are authorized to read the course instance.
+const courseInstance = await selectCourseInstance({
+  id: course_instance_id,
+  requiredRoleOptions: 'Student',
+  authzData: res.locals.authz_data,
 });
```

Comment:
Presumably this won't actually replace `res.locals.course_instance` on most existing pages, right?

If it does, that means we're doing double (unnecessary) database work on each request.

Reply:
This will not replace `res.locals.course_instance` on most pages.